File: 3408-count-the-number-of-special-characters-i/count-the-number-of-special-characters-i.py
class Solution:
    def numberOfSpecialChars(self, word: str) -> int:
        count = 0
        seen = set()
        for ch in word:
            if ch not in seen and ch.islower():
                if ch.upper() in word:
                    count += 1

            seen.add(ch)

        return count

File: 3408-count-the-number-of-special-characters-i/count-the-number-of-special-characters-i.cs
public class Solution {
    public int NumberOfSpecialChars(string word) {
        HashSet<char> lower = new HashSet<char>();
        HashSet<char> upper = new HashSet<char>();

        foreach (char ch in word) {
            if (char.IsLower(ch)) {
                lower.Add(ch);
            }
            else {
                upper.Add(char.ToLower(ch));
            }
        }

        lower.IntersectWith(upper);

        return lower.Count;
    }
}